File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/consumer/concurrent/ProductClient.java
package io.micronaut.rabbitmq.docs.consumer.concurrent;

import io.micronaut.context.annotation.Requires;
import io.micronaut.rabbitmq.annotation.Binding;
import io.micronaut.rabbitmq.annotation.RabbitClient;
// end::imports[]

@Requires(property = "spec.name", value = "ConcurrentSpec")
// tag::clazz[]
@RabbitClient // <1>
public interface ProductClient {

    @Binding("product") // <2>
    void send(byte[] data); // <3>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/consumer/concurrent/ProductClient.kt
package io.micronaut.rabbitmq.docs.consumer.concurrent

import io.micronaut.context.annotation.Requires
import io.micronaut.rabbitmq.annotation.Binding
import io.micronaut.rabbitmq.annotation.RabbitClient

@Requires(property = "spec.name", value = "ConcurrentSpec") // tag::clazz[]
@RabbitClient
interface ProductClient {

    @Binding("product")
    fun send(data: ByteArray?)
}
